pioinstaller/core.py

```diff
@@ -141,12 +144,29 @@ def check(dev=False, auto_upgrade=False, version_requirements=None):
         )
 
     try:
-        # pylint: disable=bad-option-value, import-outside-toplevel, unused-import, import-error, unused-variable, cyclic-import
-        import platformio
-    except ImportError:
-        raise exception.InvalidPlatformIOCore("Could not import PlatformIO module")
+        pio_state = fetch_platformio_state(python_exe)
+    except subprocess.CalledProcessError as e:
+        error = e.output.decode()
+        raise exception.InvalidPlatformIOCore(
+            "Could not import PlatformIO module. Error: %s" % error
+        )
+
+    pio_version = convert_version(pio_state.get("core_version"))
+    dev = dev or bool(pio_version.prerelease)
+    pio_state.update(
+        {
+            "core_dir": get_core_dir(),
+            "cache_dir": get_cache_dir(),
+            "penv_dir": penv.get_penv_dir(),
+            "penv_bin_dir": penv.get_penv_bin_dir(),
+            "platformio_exe": platformio_exe,
+            "installer_version": __version__,
+            "python_exe": python_exe,
+            "system": util.get_systype(),
+            "is_develop_core": dev,
```

Comment:
Add `python_version`. We should have it now?

Reply:
I got it from fetch_python_state.

Reply:
Cool, use it.